source/AI.cpp

```diff
@@ -1090,6 +1090,13 @@ shared_ptr<Ship> AI::FindTarget(const Ship &ship) const
 	if(!gov || ship.GetPersonality().IsPacifist())
 		return target;
 	
+	// Ships with 'plunders' personality always destroy the ships they have boarded.
+	const Personality &person = ship.GetPersonality();
+	shared_ptr<Ship> oldTarget = ship.GetTargetShip();
+	if(oldTarget && oldTarget->IsTargetable() && oldTarget->IsDisabled() && person.Plunders()
+			&& !person.Disables() && Has(ship, oldTarget, ShipEvent::BOARD))
+		return oldTarget;
+	
```

Comment:
Why do this all the way up here? I'd move these first two lines back to where they were, then put the plunders check below the timid check. (This also lets us drop the `oldTarget->IsTargetable()` part since that's done below. We know from lines 1115-1116 that for anything below that point if oldTarget exists, oldTarget is targetable.)